tests/integration/test_api_gateway.py

```diff
@@ -1193,17 +1192,24 @@ def _prepare_method_integration(
         # STEP 4: test integration with synchronous execution and response templates
 
         resp_templates = {APPLICATION_JSON: "$input.path('$.output')"}
-        _prepare_method_integration(resp_templates=resp_templates, overwrite=True, exec_async=False)
+        _prepare_method_integration(
+            resp_templates=resp_templates, overwrite=True, action="StartSyncExecution"
+        )
 
         # invoke stepfunction via API GW, assert results
         test_data_1["name"] += "2"
         resp = requests.post(url, data=json.dumps(test_data_1))
         self.assertEqual(200, resp.status_code)
         self.assertEqual(test_data, json.loads(to_str(resp.content.decode())))
 
+        _prepare_method_integration(overwrite=True, action="DeleteStateMachine")
+
+        # Remove state machine with API GW
+        resp = requests.post(url, data=json.dumps({"stateMachineArn": sm_arn}))
+        self.assertEqual(200, resp.status_code)
+
         # Clean up
         lambda_client.delete_function(FunctionName=fn_name)
-        sfn_client.delete_state_machine(stateMachineArn=sm_arn)
```

Comment:
Out of curiosity - was there an issue with this call, any particular reason why it has been removed?

Reply:
The state machine is deleted via API Gateway to test other integrations different to startexecution, clearly we cannot eliminate twice the same state machine

 https://github.com/localstack/localstack/pull/4448/files#diff-b0a7d5551e1c6d1516c3cbfb62572f4b70e45bbef9113560054a0cc702c1a886R1204-R1209

Reply:
Ah, of course - my bad. Thanks for clarifying!